src/main/java/com/google/cloud/spanner/pgadapter/ProxyServer.java

```diff
@@ -206,6 +210,13 @@ public int getNumberOfConnections() {
     return this.handlers.size();
   }
 
+  public int getLocalPort() {
+    if (this.serverSocket != null) {
+      return this.serverSocket.getLocalPort();
```

Comment:
Return the actual port that has been assigned to the socket. As we now allow port number 0 (zero) to be configured, the port that is used by the socket can have been dynamically assigned by the OS.